src/content/1.7/development/Multistore/shops-configuration/_index.md

```diff
@@ -0,0 +1,55 @@
+---
+title: Shops configuration
+menuTitle: Shops configuration
+summary: "Fetching shop configuration values in multistore context, using the ShopConstraint value object"
+weight: 30
+---
+
+# Fetching shop configuration values in multistore context
+
+When multistore is disabled and you fetch a configuration value, you are in a **single store context**, so you are fetching configuration values of your only store. All you have to do is use the **legacy configuration adapter** (more on that later).
+
+But what if you want to fetch a configuration value with more than one activated shop? This is where the **ShopConstraint** value object (VO) comes in handy, this is what you will use in order to specify which store(s) configuration value you are targeting.
+
+{{% notice note %}}
+If you don't know what is a value object, you will find out about it and its use in the [value objects]({{< ref "/1.7/development/architecture/domain/value_objects" >}}) documentation.
+{{% /notice %}}
+
+## Usage
+
+In order to fetch a configuration value in multistore context, you will need the legacy configuration adapter service `prestashop.adapter.legacy.configuration` and a ShopConstraint instance `PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint`.
+
+Example:
+
+```php
+<?php
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+
+class MyController
+{
+    // ...
+
+    public function myAction()
+    {
+        // ...
+
+        // Get the legacy configuration adapter service
+        $configuration = $container->get('prestashop.adapter.legacy.configuration');
+
+        // Instantiate a ShopConstraint value object
+        $shopConstraint = new ShopConstraint(
+            (int) $order->id_shop,
+            (int) $order->id_shop_group
+        );
+
+        // Get the needed configuration value, passing your ShopConstraint object as a third parameter
+        $roundType = (int) $configuration->get('PS_ROUND_TYPE', null, $shopConstraint);
+
+        // ...
+    }
+}
+```
+
+{{% notice note %}}
+Note that the ShopConstraint VO is not solely dedicated to being used with the configuration service. If you write a method/function that should have different behaviors depending on a given shop / shop group, then using the ShopConstraint VO is a good idea.
+{{% /notice %}}
```

Comment:
I don't know if there are some changes with the future versions of PrestaShop, but it would be worth mentioning what's happening under the hood when a configuration value is fetched.

As far as I understand, PrestaShop will check the value exist for the given shop (or group), or use on the global value.

Reply:
hmm, I could make some kind of table explaining different cases, which values are fetch in priority

Reply:
That'd be useful, that's the main feedback I hear from modules dev regarding multistore configuration data

Reply:
I added some explanation at this subject, don't hesitate to tell me if it's not so understandable. It's easy to understand but not so easy to explain :D I hope it's clear.